kasa/smartcamera.py

```diff
@@ -0,0 +1,130 @@
+"""Module for SmartCameras (EC70)."""
+from typing import Any, Dict, Optional
+
+from .credentials import Credentials
+from .modules import PTZ, Cloud, Time
+from .modules.ptz import Direction, Position
+from .smartcameraprotocol import SmartCameraProtocol
+from .smartdevice import DeviceType, SmartDevice, requires_update
+
+
+class SmartCamera(SmartDevice):
+    """Representation of a TP-Link Kasa Camera.
+
+    To initialize, you have to await :func:`update()` at least once.
+    This will allow accessing the properties using the exposed properties.
+
+    All changes to the device are done using awaitable methods,
+    which will not change the cached values, but you must await :func:`update()`
+    separately.
+
+    Errors reported by the device are raised as
+    :class:`SmartDeviceExceptions <kasa.exceptions.SmartDeviceException>`, and should be
+    handled by the user of the library.
+
+    Examples:
+        >>> import asyncio
+        >>> camera = KasaCam("127.0.0.1")
+        >>> asyncio.run(camera.update())
+        >>> print(camera.alias)
+        Camera2
+
+        Cameras, like any other supported devices, can be turned on and off:
+
+        >>> asyncio.run(camera.turn_off())
+        >>> asyncio.run(camera.turn_on())
+        >>> asyncio.run(camera.update())
+        >>> print(camera.is_on)
+        True
+    """
+
+    def __init__(
+        self,
+        host: str,
+        credentials: Credentials,
+        *,
+        port: Optional[int] = None,
+        timeout: Optional[int] = None,
+    ) -> None:
+        super().__init__(host, port=port, credentials=credentials, timeout=timeout)
+        self._device_type = DeviceType.Camera
+        self.protocol = SmartCameraProtocol(
+            host, credentials=credentials, port=port, timeout=timeout
+        )
+        self.add_module("cloud", Cloud(self, "smartlife.cam.ipcamera.cloud"))
+        self.add_module("ptz", PTZ(self, "smartlife.cam.ipcamera.ptz"))
+        self.add_module("time", Time(self, "smartlife.cam.ipcamera.dateTime"))
+
+    def _create_request(
+        self,
+        target: str,
+        cmd: str,
+        arg: Optional[Dict] = None,
```

Comment:
If the Kasa Cameras requires {} in place of "None", why not just put that as the default?  I'm not a super Python expert, so open to being wrong here if I misunderstand.
```suggestion
        arg: Optional[Dict] = {},
```

Reply:
Generally speaking it is a good idea to avoid passing mutable objects as parameters, as they are initialized at the time of declaration. I.e., if the method body would modify the arg somehow, all future calls would default to having that modified dictionary as input.

In this case it does not make any difference, but given this function overloads the base class one, it should keep using the same signature.